test/pylib/log_browsing.py

```diff
@@ -100,3 +100,17 @@ async def grep(self, expr: str | re.Pattern, filter_expr: Optional[str | re.Patt
                     matchings.append((line, m))
                 line = await self._run_in_executor(f.readline, loop=loop)
         return matchings
+
+    async def read_lines(self, from_mark: int = 0, to_mark: int = 0) -> list[str]:
+        """
+        Returns a list of lines from the Scylla log file that are between provided marks.
+        """
+        lines: list[str] = []
```

Comment:
The log file may be large, I'm afraid in some cases it may not fit into memory

Reply:
The function could be easily modified to return a generator of lines, which you could then use at call site to write line by line